src/dyad/core/dyad_core.c

```diff
@@ -511,10 +522,14 @@ DYAD_CORE_FUNC_MODS dyad_rc_t dyad_cons_store (const dyad_ctx_t* restrict ctx,
     rc = DYAD_RC_OK;
 
 pull_done:
+    DYAD_PERF_REGION_END (ctx->perf_handle, "dyad_write_data_for_consumer");
+    if (file_data != NULL) {
+        ctx->dtl_handle->return_buffer (ctx->dtl_handle, (void**)&file_data);
+    }
```

Comment:
file_data deallocation has been moved after "consume_done" in `dyad_consume()` in the latest code base.